primitives/runtime-interface/proc-macro/src/utils.rs

```diff
@@ -67,12 +108,25 @@ pub fn create_exchangeable_host_function_ident(name: &Ident) -> Ident {
 }
 
 /// Create the host function identifier for the given function name.
-pub fn create_host_function_ident(name: &Ident, trait_name: &Ident) -> Ident {
+pub fn create_host_function_ident(name: &Ident, version: u32, trait_name: &Ident) -> Ident {
 	Ident::new(
 		&format!(
-			"ext_{}_{}_version_1",
+			"ext_{}_{}_version_{}",
 			trait_name.to_string().to_snake_case(),
 			name,
+			version,
+		),
+		Span::call_site(),
+	)
+}
+
+/// Create the host function identifier for the given function name.
+pub fn create_host_shim_function_ident(name: &Ident, version: u32) -> Ident {
```

Comment:
Can we come up with a more expressive name? 

And the docs are exactly the same as for `create_host_function_ident`